### DISCUSSION TITLE
Is it possible to program via ST Link, not USB?

### DISCUSSION THREAD

**Discussion by AidanofVT** [stm32duino/Arduino_Core_STM32#1727] [Q&A] [answered]
I have an STM32F722ZET6 embedded in a PCB. I'd like to program it with the Arduino IDE, but I can't. The 'port" menu is greyed out. All online tutorials that I can find involve either using USB out of the box, or programming the MCU with a bootloader to allow for programming via USB. This circuit board doesn't have a USB port. Is there anything I can do?

**Comment by fpistm** [comment, marked as answer]
Even if there is no com port, simply select swd upload method and it should work if STM32CubeProgrammer is available in the path.
https://github.com/stm32duino/wiki/wiki/Upload-methods#stm32cubeprogrammer

**Reply by AidanofVT** [reply]
Thank you.
